### ISSUE TITLE
Tests are making the installation to loop forever 

### ISSUE THREAD

**Issue by xxLestadxx** [Consensys-Incorporated/tessera#881]
When I tried to install the project using maven 3.6 with the command ``` mvn install ``` at some point the installation kind of "stopped" without proceeding for about 7 hours. After that I stopped it manually. When providing the clean install with no tests - ```mvn clean install -DskipTests=true ``` it was installed successfully.

**Comment by prd-fox** [comment]
Does this reproduce on every install? Or just happen once?
If you clean the project first (`mvn clean`), does it continue to happen? 
Lastly, what OS and version are you using?

Installation time varies by how powerful the machine is, but usually should not take more than 15 minutes for a full `mvn install` run.

**Comment by xxLestadxx** [comment]
It was on every ``` mvn install ```. I did not use the ```mvn clean``` before using the last command with the test skip.  

OS: ubuntu 16.04 lts, CPU: i7-4510U 2.00 x4, 64 bit with RAM of 8GB

**Comment by prd-fox** [comment]
Ok, thanks. I see this thread as well on Slack: https://go-quorum.slack.com/archives/C68NY0QQZ/p1568024356139900

Looking at the `CmdLineExecutorTest`, there is one possible culprit: `directoryStoreAndSqliteWithLotsOfFilesWorks`. This test reads/writes LOTS of files, so it may be slow on systems with slow read/write speeds. Still, I would not have expected it to stall for 7 hours. I'll see if we can reduce the strain the the test causes (if it is the cause).

**Comment by xxLestadxx** [comment]
It could also be the case that it would wait for some files but they never came, like some strange deadlock, because, if I recall correctly, my CPU didn't show any processes being done, apart from the usual.

**Comment by prd-fox** [comment]
I managed to replicate this inside a Vagrant environment.

MacOS has a limit of ~10000 file descriptors, and I assume Travis set some low limit on the machines, hence why we never caught this before. Inside the vagrant environment, the limit is set to ~1 million, which isn't the most horrendous thing, but still enough to cause it to take a long time.

I'll investigate whether some sort of deadlock has indeed happened, or if its just a cause of taking a long time to convert all the files.

**Comment by prd-fox** [comment]
Are you able to report back the output of `ulimit -Ln`? This will tell us the hard limit on the number of file descriptors the OS allows.

**Comment by xxLestadxx** [comment]
```
~/$ ulimit -Ln
bash: ulimit: -L: invalid option
ulimit: usage: ulimit [-SHabcdefilmnpqrstuvxT] [limit]
```
what am I doing wrong ?

**Comment by prd-fox** [comment]
oops, my bad with a typo. It should be `ulimit -Hn`

**Comment by xxLestadxx** [comment]
```
$ ulimit -Hn
1048576
```

**Comment by prd-fox** [comment]
Thanks @xxLestadxx. We've seen a few reports from other sources for the same issue, and with the same file handle limit.
I recreated inside a Vagrant VM, with that same limit, and the test took 20 minutes to run. Although it was a VM, it is still quite a powerful machine, so machines with lower specs will take even longer.

I opened a pull request to bypass the test for high limits. Our local development and the Travis CI limit as both lower than the bypass requisite, so we still make sure the coverage is kept.

Thanks for reporting!
